packages/video_player_videohole/tizen/src/video_player.cc

```diff
@@ -105,6 +106,34 @@ int64_t VideoPlayer::Create(const std::string &uri, int drm_type,
     }
   }
 
+  if (!http_headers.empty()) {
+    auto iter = http_headers.find(flutter::EncodableValue("Cookie"));
+    if (iter != http_headers.end()) {
+      if (std::holds_alternative<std::string>(iter->second)) {
+        std::string cookie = std::get<std::string>(iter->second);
+        ret =
+            player_set_streaming_cookie(player_, cookie.c_str(), cookie.size());
+        if (ret != PLAYER_ERROR_NONE) {
+          LOG_ERROR("[MediaPlayer] player_set_streaming_cookie failed: %s.",
+                    get_error_message(ret));
+        }
+      }
+    }
+
+    iter = http_headers.find(flutter::EncodableValue("User-Agent"));
+    if (iter != http_headers.end()) {
+      if (std::holds_alternative<std::string>(iter->second)) {
+        std::string user_agent = std::get<std::string>(iter->second);
+        ret = player_set_streaming_user_agent(player_, user_agent.c_str(),
+                                              user_agent.size());
+        if (ret != PLAYER_ERROR_NONE) {
+          LOG_ERROR("[MediaPlayer] player_set_streaming_user_agent failed: %s.",
+                    get_error_message(ret));
+        }
+      }
+    }
+  }
```

Comment:
suggest extract a new method to get value from http headers, for example:
```cpp
template <typename T>
void ExtractValueFromMap(const flutter::EncodableValue &arguments,
                         const char *key, T &outValue) {
  if (std::holds_alternative<flutter::EncodableMap>(arguments)) {
    flutter::EncodableMap values = std::get<flutter::EncodableMap>(arguments);
    flutter::EncodableValue value = values[flutter::EncodableValue(key)];
    if (std::holds_alternative<T>(value)) outValue = std::get<T>(value);
  }
}
```

Reply:
fixed